# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/pcem/hotfix.patch.arm64
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

I tried, bu their custom cmake/arch-detect is too broken and cursed
for me to spend more time tweeking. Just set it so it is not unknown
and builds x86 code nontheless. -mstackrealign is also unsupported.
Maybe all that can be made generic anyway.

--- ./CMakeLists.txt.vanilla	2024-02-12 21:08:52.180006632 +0100
+++ ./CMakeLists.txt	2024-02-12 21:09:37.368006653 +0100
@@ -8,15 +8,16 @@
 include(${CMAKE_SOURCE_DIR}/includes/includes.cmake)
 include(${CMAKE_SOURCE_DIR}/cmake/arch-detect.cmake)
 
-target_architecture(PCEM_CPU_TYPE)
+#target_architecture(PCEM_CPU_TYPE)
+set(PCEM_CPU_TYPE "arm64")
 
 if(${PCEM_CPU_TYPE} STREQUAL "i386")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse2")
 endif()
 
-set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -fomit-frame-pointer -mstackrealign -fno-strict-aliasing")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -fomit-frame-pointer -mstackrealign -fno-strict-aliasing")
+set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -fomit-frame-pointer -fno-strict-aliasing")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -fomit-frame-pointer -fno-strict-aliasing")
 
 set(PCEM_VERSION_STRING "vNext" CACHE STRING "PCem Version String")
 set(PCEM_DISPLAY_ENGINE "wxWidgets" CACHE STRING "PCem Display Engine")
